### ISSUE TITLE
Compiling on aarch64: immintrin.h not found

### ISSUE THREAD

**Issue by parkerlreed** [suijingfeng/vkQuake3#14]
It exists in clang but isnt used in gcc for ARM. Seems to be a x86 specific file.

```
[parker@alarm games]$ find /usr/lib -name immintrin.h
/usr/lib/clang/10.0.1/include/immintrin.h
```

```
[parker@alarm vkQuake3]$ make -j4                
make[1]: Entering directory '/home/parker/build/vkQuake3'

Building in build/release-linux-aarch64:
  PLATFORM: linux
  ARCH: aarch64
  VERSION: 1.36_GIT_38062bbf-2020-07-07
  COMPILE_PLATFORM: linux
  COMPILE_ARCH: aarch64
  HAVE_VM_COMPILED: 
  PKG_CONFIG: pkg-config
  CC: cc

  CFLAGS:
    -Wall -fno-strict-aliasing -Wimplicit -Wstrict-prototypes -pipe -DUSE_ICON -DARCH_STRING="aarch64"
    -DNO_VM_COMPILED -DNO_GZIP -Icode/zlib -DUSE_INTERNAL_JPEG -Icode/jpeg-8c -DUSE_LOCAL_HEADERS
    -DPRODUCT_VERSION="1.36_GIT_38062bbf-2020-07-07" -Wformat=2 -Wno-format-zero-length -Wformat-security
    -Wno-format-nonliteral -Wstrict-aliasing=2 -Wmissing-format-attribute -Wdisabled-optimization
    -Werror-implicit-function-declaration -MMD -DNDEBUG -O3 -ffast-math

  CLIENT_CFLAGS:
    -I/usr/include/SDL2 -D_REENTRANT -DUSE_OPENAL -DUSE_OPENAL_DLOPEN -DUSE_CURL -DUSE_CURL_DLOPEN -DUSE_VOIP
    -DUSE_CODEC_OPUS -DOPUS_BUILD -DHAVE_LRINTF -DFLOATING_POINT -DFLOAT_APPROX -DUSE_ALLOCA -Icode/opus-1.2.1/include
    -Icode/opus-1.2.1/celt -Icode/opus-1.2.1/silk -Icode/opus-1.2.1/silk/float -Icode/opusfile-0.9/include
    -DUSE_CODEC_VORBIS -Icode/libvorbis-1.3.6/include -Icode/libvorbis-1.3.6/lib -Icode/libogg-1.3.3/include
    -DUSE_RENDERER_DLOPEN -DUSE_MUMBLE

  SERVER_CFLAGS:
    -DUSE_VOIP

  LDFLAGS:
    

  LIBS:
    -ldl -lm

  CLIENT_LIBS:
    -pthread -lSDL2 -lrt

  AUTOUPDATER_LIBS:
    -ldl

  Output:
    ioq3ded.aarch64
    ioquake3.aarch64
    renderer_opengl1_aarch64.so
    renderer_opengl2_aarch64.so
    renderer_vulkan_aarch64.so
    baseq3/cgameaarch64.so
    baseq3/qagameaarch64.so
    baseq3/uiaarch64.so
    missionpack/cgameaarch64.so
    missionpack/qagameaarch64.so
    missionpack/uiaarch64.so

make[2]: Entering directory '/home/parker/build/vkQuake3'
LD build/release-linux-aarch64/ioq3ded.aarch64
CC code/client/cl_main.c
CC code/client/cl_net_chan.c
CC code/client/cl_parse.c
CC code/client/cl_scrn.c
CC code/client/cl_ui.c
In file included from code/client/../sys/../SDL2/include/SDL.h:38,
                 from code/client/../sys/sys_loadlib.h:39,
                 from code/client/cl_main.c:28:
code/client/../sys/../SDL2/include/SDL_cpuinfo.h:86:10: fatal error: immintrin.h: No such file or directory
   86 | #include <immintrin.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:2794: build/release-linux-aarch64/client/cl_main.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/parker/build/vkQuake3'
make[1]: *** [Makefile:1407: targets] Error 2
make[1]: Leaving directory '/home/parker/build/vkQuake3'
make: *** [Makefile:1323: release] Error 2
```

**Comment by parkerlreed** [comment]
I edited out the immintrin.h include from https://github.com/suijingfeng/vkQuake3/blob/master/code/SDL2/include/SDL_cpuinfo.h

And it seems to be compiling fine now.

**Comment by parkerlreed** [comment]
Compiled fine and runs great. Is there a less hacky way around this?

**Comment by suijingfeng** [comment]
immintrin.h is only for x86-64, About SSE, it seems that this is SDL's bug, either update SDL headers or you solve this problem and report it to the author of SDL library. I am not test this game on ARM, not have a arm cpu, sorry.

**Comment by br6** [comment]
> immintrin.h is only for x86-64, About SSE, it seems that this is SDL's bug, either update SDL headers or you solve this problem and report it to the author of SDL library. I am not test this game on ARM, not have a arm cpu, sorry.

what did you edit 

> I edited out the immintrin.h include from https://github.com/suijingfeng/vkQuake3/blob/master/code/SDL2/include/SDL_cpuinfo.h
> 
> And it seems to be compiling fine now.

Hello,

what was edited?

**Comment by parkerlreed** [comment]
Removed 85 and 86 https://github.com/suijingfeng/vkQuake3/blob/master/code/SDL2/include/SDL_cpuinfo.h#L85

(And also one of the endfis if necessary, I don't recall)

**Comment by perkele1989** [comment]
I solved this by building with  `SDL_DISABLE_IMMINTRIN_H`

```
CFLAGS="-O2 -march=armv8-a+crc+simd -mtune=cortex-a72 -DSDL_DISABLE_IMMINTRIN_H" CXXFLAGS="-O2 -march=armv8-a+crc+simd -mtune=cortex-a72 -DSDL_DISABLE_IMMINTRIN_H"  make -j4
```

**Comment by parkerlreed** [comment]
@haikarainen Thanks, great to know!

**Comment by suijingfeng** [comment]
yes, have the same problem when build on mips cpu

**Comment by suijingfeng** [comment]
fixed by ec8089b0940b973fab28b88267b1571af4ae9b3f, @parkerlreed  @haikarainen  @br6  @mpalmer
